### PR TITLE
[optimize] update activity organization infomation

### LINKED ISSUE #136: 主办方信息 SEO
## 活动详情页的主办方信息页放在单独的页面内

- 利于 SEO
- 最好采用 Next.js 官方推荐服务端渲染写法

### Task

- [ ] SSR
- [x] 前端增加 Url 字段 https://github.com/kaiyuanshe/open-hackathon-api/pull/339

### LINKED ISSUE #137: 活动详情主办方信息为空
Image

## 解决方案 （任选其一）

1. 管理员创建活动时必须添加主办方信息
2. 主办方信息为空时不显示

### PR DESCRIPTION
**Pull request by wangrunlin**
Checklist（清单）:

- [x] Labels
- [ ] Assignees
- [x] Reviewers

Closes #136 
Closes #137

### CODE REVIEW

**Comment by wangrunlin** [comment]
@TechQuery 测试服麻烦给一个已上线的活动的管理权限

File: locales/zh-Hant-TW.json [hunk 1]
```diff
@@ -182,5 +182,7 @@
   "content": "內容",
   "sure_delete_this_message": "確定刪除該公告？",
   "admin_management": "管理管理員",
-  "confirm_to_delete_platform_admin": "確定刪除所选平台管理員？"
+  "confirm_to_delete_platform_admin": "確定刪除所选平台管理員？",
+  "please_enter_url": "請輸入 URI",
```
Comment (TechQuery) [inline locales/zh-Hant-TW.json:186]:
```suggestion
  "please_enter_url": "請輸入 URL",
```

File: models/Organization.ts [hunk 2]
```diff
@@ -62,3 +63,5 @@ export class OrganizationModel extends Stream<Organization>(ListModel) {
     return body!;
   }
 }
+
+export default new OrganizationModel('');
```
Comment (TechQuery) [inline models/Organization.ts:67]:
```suggestion
```
子级接口不能导出单例，要从上级接口生成：
https://github.com/kaiyuanshe/OpenHackathon-Web/blob/982c80617cb8f74ae6296ddf5f002960ef0e649e/models/Activity.ts#L109-L113

File: components/Organization/ActivityOrganizationModal.tsx [hunk 3]
```diff
@@ -102,6 +106,14 @@ export class OrganizationModal extends PureComponent<OrganizationModalProps> {
               required
             />
           </Form.Group>
+          <Form.Group className="mt-2" controlId="url">
+            <Form.Label>{t('url')}</Form.Label>
+            <Form.Control
+              name="url"
+              placeholder={t('please_enter_url')}
+              required
+            />
```
Comment (TechQuery) [inline components/Organization/ActivityOrganizationModal.tsx:115]:
```suggestion
            <Form.Control
              type="url"
              name="url"
              required
              placeholder={t('please_enter_url')}
            />
```

File: pages/activity/[name]/index.tsx [hunk 4]
```diff
@@ -26,29 +26,33 @@ import {
...
 import { convertDatetime } from '../../../utils/time';
 
 export async function getServerSideProps({
   params: { name = '' } = {},
 }: GetServerSidePropsContext<{ name?: string }>) {
   try {
     const activity = await activityStore.getOne(name);
+    const organizationList = await new OrganizationModel(
+      `hackathon/${name}`,
+    ).getList();
```
Comment (TechQuery) [inline pages/activity/[name]/index.tsx:47]:
```suggestion
    const organizationList = await activityStore.organizationOf(name).getList();
```

File: pages/activity/[name]/index.tsx [hunk 5]
```diff
@@ -287,8 +289,18 @@ export default class ActivityPage extends PureComponent<
             </Tabs>
           </Col>
           <Col className="d-flex flex-column">
-            <h2>{t('sponsor_information')}</h2>
-            <OrganizationList store={this.organizationStore} />
+            {this.props.organizationList.length > 0 && (
```
Comment (TechQuery) [inline pages/activity/[name]/index.tsx:292]: 上方解构 `organizationList`

File: pages/activity/[name]/index.tsx [hunk 6]
```diff
@@ -287,8 +289,18 @@ export default class ActivityPage extends PureComponent<
...
+              <>
+                <h2>{t('sponsor_information')}</h2>
+                <ul className="list-unstyled">
+                  {this.props.organizationList.map(item => (
+                    <li key={item.id}>
+                      <OrganizationCard {...item} />
+                    </li>
+                  ))}
+                </ul>
```
Comment (TechQuery) [inline pages/activity/[name]/index.tsx:301]: 用现成组件不行吗？

https://github.com/kaiyuanshe/OpenHackathon-Web/blob/982c80617cb8f74ae6296ddf5f002960ef0e649e/components/Organization/OrganizationList.tsx#L18-L28

**Review by TechQuery — CHANGES_REQUESTED** [review]

**Review by TechQuery — APPROVED** [review]

**Pull request merged**